workflow/controller/steps.go

```diff
@@ -93,7 +93,11 @@ func (woc *wfOperationCtx) executeSteps(nodeName string, tmplCtx *templateresolu
 
 		sgNode := woc.executeStepGroup(stepGroup.Steps, sgNodeName, &stepsCtx)
 
-		if !sgNode.Fulfilled() {
+		if sgNode.Fulfilled() {
```

Comment:
If the step template has mutex lock and will not invoke execute template to release the lock if child nodes are completed.  It. has to be handled here
Eg:
```yaml
- name: main-function
    synchronization:
      mutex:
        name:  config
    steps:
    - - name: check
        template: main-command
    - - name: check1
        template: main-command


```